proto/babylon/incentive/genesis.proto

```diff
@@ -3,10 +3,48 @@ package babylon.incentive;
 
 import "gogoproto/gogo.proto";
 import "babylon/incentive/params.proto";
+import "babylon/incentive/incentive.proto";
+import "cosmos_proto/cosmos.proto";
 
 option go_package = "github.com/babylonlabs-io/babylon/x/incentive/types";
 
 // GenesisState defines the incentive module's genesis state.
 message GenesisState {
-    Params params = 1 [(gogoproto.nullable) = false];
+  // params the current params of the state.
+  Params params = 1 [ (gogoproto.nullable) = false ];
+  // BTC staking gauge on every height
+  repeated BTCStakingGaugeEntry btc_staking_gauges = 2
+      [ (gogoproto.nullable) = false ];
+  // Staking Rewards gauge on every height
+  repeated RewardGaugeEntry reward_gauges = 3 [ (gogoproto.nullable) = false ];
 }
+
+// BTCStakingGaugeEntry represents a gauge for BTC staking rewards at a specific
+// height.
+message BTCStakingGaugeEntry {
+  // Block height at which this gauge is set
+  uint64 height = 1;
+  // The gauge object storing rewards info
+  Gauge gauge = 2;
+}
+
+// RewardGaugeEntry represents a reward gauge for a specific stakeholder.
+message RewardGaugeEntry {
+  // Type of stakeholder
+  StakeholderType stakeholder_type = 1;
+  // Address of the stakeholder
+  string address = 2 [ (cosmos_proto.scalar) = "cosmos.AddressString" ];
+  // The reward gauge object
+  RewardGauge reward_gauge = 3;
+}
+
+// StakeholderType represents the different types of stakeholders.
+enum StakeholderType {
```

Comment:
this is a dangerous part :sweat_smile:  we should definitely do it! 
Just needs to verify if a migration is needed in the upgrade handlers
https://github.com/babylonlabs-io/babylon/blob/ab5129460681b4682f44ef66bdc764ba7060f7ff/app/upgrades/README.md#L1

Reply:
yeah, good point!
I'll test if this breaks. In such case, we'll need to add the corresponding migration.

Reply:
tested this and works OK

Reply:
how did you did this test?

Reply:
I started a local chain with the current changes with some reward gauge seeds on the genesis. Then switched to the bin on main without these changes and queried the reward gauges. Also just added a [test here](https://github.com/babylonlabs-io/babylon/blob/d6de2fc6406a0664cbae6331a48ee79eb36cad1c/x/incentive/keeper/reward_gauge_test.go#L44) to make sure it is non-breaking. Let me know if we need a more thorough test

Reply:
awesome! that makes sense
Thanks!

Reply:
I think it is not breaking as we are stil using the same method to convert enum to bytes i.e:
```
func (st StakeholderType) Bytes() []byte {
	return []byte{byte(st)}
}
```
If we would switch to proto serialization then most probably it would be breaking change.